### PR TITLE
storage: Fix links and hovering

### PR DESCRIPTION
**Pull request by garrett**
Follow-up to https://github.com/cockpit-project/cockpit/pull/15903

### CODE REVIEW

**Comment by garrett** [comment]
Everything but the last three commits are handled in the other PR. Once that lands and this is rebased, those should drop out. (However, the commits depend on the others, so they're here as well.)

**Comment by garrett** [comment]
We chatted about this in IRC today. The main points:

- The goal of this PR was to make different imeplemtations of similar widgets act similarly. Right now, there are multiple versions of how this is shown in the Storage page.
- PatternFly's hover state is lacking. It doesn't look clickable. It's too subtle and in many cases (bad laptop screen, poor room lighting, multiple sight-related disabilities), the light grey might not even be visible.
- The list hover action in Cockpit was created to solve issues found in UX studies, to make them look more clickable. The hover with a link solved the issues when re-testing.
- Relying on color alone is problematic due to some forms of colorblindness.
- Making every item look like a link is overkill and becomes noisy.
- Perhaps using a PF4 list with links on the titles of each item (expanded to the cell to increase the clickable target) might be the best solution. It's either this or making the whole row clickable and show link style on hover, like in the journal and in this PR in commit 9e6b09f.

**Comment by garrett** [comment]
Rebased.

File: pkg/storaged/drives-panel.jsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ export class DrivesPanel extends React.Component {
                               name={name}
                               devname={block_name(block)}
                               detail={desc}
-                              go={() => cockpit.location.go([dev])}
+                              location={"#/" + dev}
```
Comment (mvollmer) [inline pkg/storaged/drives-panel.jsx:89]: We should pass the original argument here: `location={[dev]}` and use `cockpit.location.go` in SidePanelRow below, to get the encoding right.  If we really need to use `href` of the Button, let's use `cockpit.location.encode` to encode the abstract location into a URL fragment.

**Review by mvollmer — CHANGES_REQUESTED** [review]

File: pkg/storaged/drives-panel.jsx (hunk 1 above)
Comment (KKoukiou) [reply pkg/storaged/drives-panel.jsx:89]: Thanks for taking a look. button links should use href for accessibility purposes  indeed - but here we haven't finalized yet if we should actually use links. But thanks for the pointer!

**Comment by garrett** [comment]
> if we should actually use links

We probably should. And if they're block (which expand to the parent cell), they would be easier to click/tap than inline. (This is what we do on other pages, like Machines and Services top-level.

Other devices would need to be fixed to have a title, I guess. (Not sure what the title would be though.)

**Comment by garrett** [comment]
Closed, in preference to #16033.

We'll eventually want to upgrade the lists on the page, but later.

**Pull request closed without merging**